lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -537,52 +537,88 @@ class AwsCompileFunctions {
         versionCompilationPromise = BbPromise.resolve();
       }
 
-      return versionCompilationPromise.then(() => this.compileFunctionDestinations(functionName));
+      return versionCompilationPromise.then(() =>
+        this.compileFunctionEventInvokeConfig(functionName)
+      );
     });
   }
 
-  compileFunctionDestinations(functionName) {
+  compileFunctionEventInvokeConfig(functionName) {
     const functionObject = this.serverless.service.getFunction(functionName);
-    const { destinations } = functionObject;
-    if (!destinations) return;
+    const {
+      destinations,
+      maximumEventAge: maximumEventAgeOnFunction,
+      maximumRetryAttempts: maximumRetryAttemptsOnFunction,
+    } = functionObject;
+    let maximumEventAgeOnDestinations;
+    let maximumRetryAttemptsOnDestinations;
+
+    if (!destinations && !maximumEventAgeOnFunction && !maximumRetryAttemptsOnFunction) return;
+
+    const destinationConfig = {};
+
+    if (destinations) {
+      if (destinations.maximumRetryAttempts !== undefined) {
+        this.serverless._logDeprecation(
+          'AWS_FUNCTION_DESTINATIONS_MAXIMUM_RETRY_ATTEMPTS',
+          'destinations.maximumRetryAttempts is deprecated, use maximumRetryAttempts on function instead'
+        );
+        maximumRetryAttemptsOnDestinations = destinations.maximumRetryAttempts;
+      }
+
+      if (destinations.maximumEventAge !== undefined) {
```

Comment:
Here as all supported values are not falsy, check can be even simpler `if (destinations.maximumEventAge)`